admin-notice/class-admin-notice-controller.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Automattic\VIP\Admin_Notice;
+
+// function sample_admin_notice__error() {
+//     $class = 'notice notice-info';
+//     $message = __( 'Irks! An error has occurred.', 'sample-text-domain' );
+
+//     printf( '<div class="%1$s"><p>%2$s</p></div>', esc_attr( $class ), esc_html( $message ) );
+// }
+// add_action( 'admin_notices', 'sample_admin_notice__error' );
+
+
+require_once __DIR__ . '/class-admin-notice.php';
+
+
+class Admin_Notice_Controller {
+	private $notice_class = 'notice notice-info';
+	public $all_notices = [];
+
+	public static function init() {
+		$instance = new Admin_Notice_Controller();
+
+		add_action( 'admin_notices', [ $instance, 'display_notices' ] );
+
+		$instance->all_notices = [
+			new Admin_Notice( 'WordPress 5.5.2 will be released on Friday, October 30th', '01-07-2020', '30-10-2020 15:00' ),
```

Comment:
Not sure if this is the best way of defining the notices? 

I was thinking about having them in separate file, but I am not sure if I should split it either. What do you think?

Reply:
Seems like the list of notices should really be defined elsewhere, and not buried in a class of what should be re-usable logic.

Reply:
Instead of `Admin_Notices::init()` in the base file, you could do something like:

```
$admin_notices = new Admin_Notices();
$admin_notices->add( new Admin_Notice( 'WordPress 5.5.2 will be released on Friday, October 30th', '01-07-2020', '30-10-2020 15:00' ) );
// ... add more notices as needed ...
$admin_notices->render();
```

That would move the data about admin notices into the root of the plugin, avoid having to use a a static init factory method, uses composition instead of highly coupling Admin_Notice and Admin_Notices classes (which then allows for easier mocking in tests), and allows for easier use of decorators to limit who should be shown certain admin messages:

```
$admin_notices = new Admin_Notices();
$admin_notices->add( new Editor_Notice( new Admin_Notice( 'WordPress 5.5.2 will be released on Friday, October 30th', '01-07-2020', '30-10-2020 15:00' ) ) );
// ... add more notices as needed ...
$admin_notices->render();

...

// Should be called Admin_Notice, but using suffix here to make it clearer for the existing code. Concrete class could be Info_Notice, to then allow for Error_Notice or Warning_Notice as well.
interface Admin_Notice_Interface {}

...

// Base Decorator - so override the individual Admin_Notice
class Admin_Notice_Decorator {
	protected $notice;
	public function __construct( Admin_Notice_Interface $notice ) {
		$this->notice = $notice;
	}
}

...

class Editor_Notice extends Admin_Notice_Decorator {
    public function render() {
		if ( current_user_can( 'edit_posts' ) {
			$this->notice->render();
		}
    }
}

```

Reply:
Great! I loved the idea of putting it to the main file!

About the naming, I called the main class `Admin_Notice_Controller`. It seems more fitting than `Admin_Notices` as it is not just a collection, but rather has some more logic.But that could be my Java backage background talking:) Do you think `Admin_Notices` would be a better name?

Reply:
You could have both - Notices is the collection, and Controller (if still needed) does things with the collection (like adding and removing single notices).

If the logic is moved to the root file (roughly per my example), then that's really the controlling  logic / composition root there, so may not need a separate controller class.